### DISCUSSION TITLE
Extracting API routes from `pages` directory`

### DISCUSSION THREAD

**Discussion by sejr** [vercel/next.js#11226] [Help]
Hi all,

I have been working on a rather large NextJS application and wonder if this discussion has come up before. I have a _lot_ of different pages, and am interested in moving my API routes outside of the `pages` directory to help better organize the project. Is there any way currently to do this, perhaps through `next.config.js`?

**Comment by waptik** [comment]
Use Now nodejs package like [this](https://github.com/HelloEdit/authomatic)  or try [this](https://github.com/poulainv/tottem)
